### ISSUE TITLE
Sharing drivers on multi user systems

### ISSUE THREAD

**Issue by jbrandru** [dbeaver/dbeaver#3858]
Hi

Is there anyway to share the database drivers on multi user systems so that users don't have to download them? Ive been asked to set up dbeaver on a few virtual desktops in a closed environment (ei. no internet access) but I can seem to find a way to install the drivers as the application user that will allow the normal users to use them without having to manually edit the config file for each user.

If not is this something you would consider adding in the near future 

/Jacob

**Comment by tati-kru** [comment]
Hi Jacob,
The drivers for all main databases are included to the enterprise version of DBeaver. It is a part of package of DBeaver EE for the cases exactly like yours (closed environment without internet access).

**Comment by serge-rider** [comment]
Also you can configure drivers path in preferences. It could be some network path so multiple users will be able to use it. Each DBeaver installation will still has its own config but at least it won't re-download binaries every time.

![image](https://user-images.githubusercontent.com/6398845/43679854-a3be66d4-9835-11e8-8cb2-b80f3365ea1f.png)

**Comment by windowshome** [comment]
I don't think you understand the remote desktop multi-user scenario. The accounts of each user under remote multi-user are independent. I hope there is a global setting for the driver, so that each different user doesn't need to set the driver separately to access it. Can you understand;

**Comment by eliasthecactus** [comment]
I ran into the same thing on a Windows RDS server with temporary user profiles, so I am leaving the setup here. This is working with DBeaver Community 26.1.4.

## Shared JDBC drivers

Put the JDBC jars and a `drivers.xml` somewhere machine-wide, for example:

```text
C:\ProgramData\DBeaver\drivers.xml
C:\ProgramData\DBeaver\drivers\my-driver.jar
```

The users only need read access. Then add this to the shared DBeaver `dbeaver.ini`, **after** the `-vmargs` line:

```text
-Ddbeaver.drivers.configuration-file=C:\ProgramData\DBeaver\drivers.xml
```

A small `drivers.xml` can look like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<drivers>
  <provider id="generic">
    <driver id="my-driver" name="My JDBC Driver" class="com.example.Driver" url="jdbc:example://{host}:{port}/{database}" custom="true">
      <library type="lib" path="C:\ProgramData\DBeaver\drivers\my-driver.jar" custom="true"/>
    </driver>
  </provider>
</drivers>
```

Use an existing provider such as `generic`. A made-up provider ID is ignored. DBeaver loads this central file before the user-specific `drivers.xml`, so every new RDS profile gets the driver without downloading or configuring it again.

Official docs: https://dbeaver.com/docs/dbeaver/Admin-Manage-Drivers/

## Shared settings for new users

DBeaver is Eclipse-based, so you can also provide machine-wide default preferences with `-pluginCustomization`.

Create:

```text
C:\ProgramData\DBeaver\plugin_customization.ini
```

Example contents:

```properties
# Keep the defaults from DBeaver's bundled plugin_customization.ini
org.eclipse.ui/SHOW_PROGRESS_ON_STARTUP=true
org.eclipse.ui/KEY_CONFIGURATION_ID=org.jkiss.dbeaver.defaultKeyScheme
org.eclipse.egit.ui/show_home_drive_warning=false

# Disable telemetry and skip the consent dialog
org.jkiss.dbeaver.ui.statistics/feature.tracking.enabled=false
org.jkiss.dbeaver.ui.statistics/feature.tracking.skipConfirmation=true

# Do not ask to create the sample database
org.jkiss.dbeaver.core/sample.database.canceled=true
```

Then add these two lines to `dbeaver.ini` **before** `-vmargs`:

```text
-pluginCustomization
C:/ProgramData/DBeaver/plugin_customization.ini
```

For 26.1.4 I also set this **after** `-vmargs` so the newer first-run Product Configuration wizard cannot show the same telemetry/sample options:

```text
-Ddbeaver.show.easy.config.on.startup=false
```

The important detail is that `-pluginCustomization` is before `-vmargs`, while all `-D...` properties are after it.

The preference file format is:

```text
<plugin-id>/<preference-key>=<value>
```

You can find the keys in a configured workspace under:

```text
.metadata\.plugins\org.eclipse.core.runtime\.settings\*.prefs
```

This works well for RDS/VDI because every fresh profile and workspace starts with the same defaults. These are defaults, not locked policies: an existing workspace can already have user values that override them, and a user can change them later. For temporary profiles/new workspaces that is normally exactly what is needed.

Preference docs: https://dbeaver.com/docs/dbeaver/Admin-Manage-Preferences/
